src/datachain/query/dataset.py

```diff
@@ -664,13 +1033,36 @@ class UDFSignal(UDFStep):
     min_task_size: int | None = None
     batch_size: int | None = None
 
-    def create_udf_table(self, query: Select) -> "Table":
+    def processed_input_ids_query(self, partial_table: "Table"):
+        """
+        For mappers (1:1 mapping): returns sys__id from partial table.
+
+        Since mappers have a 1:1 relationship between input and output,
+        the sys__id in the partial table directly corresponds to input sys__ids.
+        """
+        return sa.select(partial_table.c.sys__id.label("sys_id")).subquery()
```

Comment:
Mistype? 🤔 Do we have test for this?

```suggestion
        return sa.select(partial_table.c.sys__id.label("sys__id")).subquery()
```

Reply:
1. Column names in tables: `sys__id` (double underscore) - the actual column in database tables
2. Label in subquery: `sys_id` (single underscore) - an alias used in the subquery result

Here's the flow:

In UDFSignal:
```python
partial_table.c.sys__id.label("sys_id")  # Column sys__id labeled as sys_id
```

In RowGenerator:
```python
partial_table.c.sys__input_id.label("sys_id")  # Column sys__input_id labeled as sys_id
```

Then in calculate_unprocessed_rows:
```python
processed_input_ids_subquery.c.sys_id  # Access the labeled column
```

The `.label("sys_id")` creates a common interface so that both `UDFSignal` and `RowGenerator` return a subquery with the same column name (sys_id), even though they're selecting from different source columns (`sys__id` vs `sys__input_id`). This allows `calculate_unprocessed_rows()` to work with both types without knowing which subclass it's dealing with.

**The problem is though** that I choose really bad name - it's very similar to `sys__id` and it's not under `sys__` namespace which cause collision with user defined columns. Now I'm renaming it to something more reasonable - `sys__processed_id`. I know that you commented above that we shouldn't add new columns into `sys__` but this is not doing that actually, it's just "borrowing" `sys__` namespace for a subquery - it doesn't save it into DB as column.

Reply:
was it resolved? (renamed?)

Reply:
yes